lib/src/component_declaration/component_base.dart

```diff
@@ -87,7 +88,52 @@ class UiComponent2BridgeImpl extends Component2BridgeImpl {
 
   @override
   JsMap jsifyPropTypes(covariant UiComponent2 component, Map propTypes) {
-    // todo implement jsifyPropTypes
+    // Add [PropValidator]s for props annotated as required.
+    var newPropTypes = Map.from(propTypes);
+    component.consumedProps?.forEach((ConsumedProps consumedProps) {
+      consumedProps.props.forEach((PropDescriptor prop) {
+        if (!prop.isRequired) return;
+
+        Error requiredPropValidator(
+            Map props,
+            String propName,
+            String componentName,
+            String location,
+            String propFullName,
+          ) {
+            if (prop.isNullable && props.containsKey(prop.key)) return null;
+            if (!prop.isNullable && props[prop.key] != null) return null;
+
+            if (props[propName] == null) {
+              return new PropError.required(propName, prop.errorMessage);
+            }
+            return null;
+          }
+          newPropTypes[prop.key] = requiredPropValidator;
```

Comment:
Mm, what if there is a consumer prop validator as well as the required one; wouldn't the required one get overridden? Could we add a test for that?

We could either chain these two, or call it within the `requiredPropValidator`.

Reply:
oooooo good thought! golly